lib/Sema/CSDiagnostics.cpp

```diff
@@ -3729,21 +3729,20 @@ static void diagnoseUnsafeCxxMethod(SourceLoc loc, Type baseType,
         name.getBaseIdentifier().str() == "end") {
       ctx.Diags.diagnose(loc, diag::dont_use_iterator_api,
                          name.getBaseIdentifier().str());
-    } else if (cxxMethod->getReturnType()->isPointerType())
+    } else if (cxxMethod->getReturnType()->isPointerType()) {
       ctx.Diags.diagnose(loc, diag::projection_not_imported,
-                         name.getBaseIdentifier().str(), "pointer");
```

Comment:
I do actually think it's important to differentiate between pointers and references here, as C++ programmers may not immediately recognize a reference as a projection. Calling out specifically that a "reference" is returning an unsafe address is helpful, I think.